src/tree/containers/ContainerTreeItem.ts

```diff
@@ -70,13 +70,12 @@ export class ContainerTreeItem extends AzExtTreeItem {
         }
     }
 
-    public getContainer(): Container {
-        return ext.dockerode.getContainer(this.containerId);
+    public async getContainer(): Promise<Container> {
+        return callDockerode(() => ext.dockerode.getContainer(this.containerId));
```

Comment:
The rule is: use 'return await' inside `try {}` block but _do not_ use 'return await' outside of a `try {}` block. [This is why](https://jakearchibald.com/2017/await-vs-return-vs-return-await/). I am following the rule here and getting no warnings when I build.

Reply:
Huh, these two rules are contradictory (they agree on inside try-catch though).

From my link (emphasis mine):
> **Returning an awaited promise can make sense for better stack trace information** as well as for consistent error handling (returned promises will not be caught in an async function try/catch).

From your link (emphasis mine):
> Inside an `async function`, `return await` is seldom useful. Since the return value of an `async function` is always wrapped in `Promise.resolve`, **`return await` doesn’t actually do anything except add extra time before the overarching Promise resolves or rejects**. The only valid exception is if `return await` is used in a try/catch statement to catch errors from another Promise-based function.

@philliphoff Any thoughts?

Reply:
If, literally, the *only* thing a function does is call another function which returns a `Promise`, then I don't see the point or any significant benefit to `return await ...`.  I can buy the argument that, if that function does some other (synchronous) stuff first and then calls another function that returns a `Promise`, `return await ...` can make error handling in the caller easier (because the function will return only rejected `Promise` rather than either an exception or a rejected `Promise`).  Still, it's unlikely that the outer chain of callers will not be handling both cases anyway, so it's of little real value (IMO).

Reply:
Fair enough 👍